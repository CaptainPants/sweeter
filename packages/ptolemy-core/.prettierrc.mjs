export { default as default } from '../../prettierrc.mjs';
