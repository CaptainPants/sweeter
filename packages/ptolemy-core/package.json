{
    "name": "@serpentis/ptolemy-core",
    "version": "0.0.0",
    "repository": "https://github.com/serpentis-software/ptolemy",
    "author": "Serpentis Software <captainpantsnpm@gmail.com>",
    "license": "MIT",
    "private": false,
    "type": "module",
    "packageManager": "pnpm@9.15.3",
    "main": "./dist/index.js",
    "types": "./dist/index.d.ts",
    "exports": {
        ".": {
            "import": "./dist/index.js",
            "require": "./dist/index.cjs"
        }
    },
    "scripts": {
        "trace": "tsc --noEmit --incremental false --generateTrace .trace && pnpm exec analyze-trace .trace --forceMillis 20 --skipMillis 10",
        "clean": "rimraf dist; rimraf *.tsbuildinfo; rimraf .rollup.cache; :",
        "typecheck": "tsc --noEmit",
        "format": "prettier --write src",
        "format:check": "prettier --check src",
        "lint": "eslint src --report-unused-disable-directives --max-warnings 0",
        "test:watch": "vitest --globals",
        "test": "vitest --globals --watch=false",
        "verify:fix": "corepack pnpm run lint --fix && corepack pnpm run format && corepack pnpm run test && corepack pnpm run typecheck",
        "verify": "corepack pnpm run lint && corepack pnpm run test && corepack pnpm run typecheck",
        "watch": "vite build --watch",
        "build": "pnpm run typecheck && vite build"
    },
    "engines": {
        "pnpm": "9.15.3"
    },
    "dependencies": {
        "@serpentis/ptolemy-utilities": "workspace:^"
    },
    "devDependencies": {
        "@serpentis/rollup-plugin-ptolemy": "workspace:^"
    }
}
