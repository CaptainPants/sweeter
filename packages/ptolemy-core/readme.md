# Ptolemy Core

This package contains the shared core of Ptolemy. 

This includes:
- Signals
- Base Components
- Base Hooks
- Context

See [here](../../readme.md) for full Ptolemy documentation.