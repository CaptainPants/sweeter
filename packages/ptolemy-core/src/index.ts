import './jsx-globals.js';
import './PtolemyExtensionPoints.js';
import './global.js';

export * from './components/index.js';
export * from './context/index.js';
export * from './dev.js';
export * from './executionContext/index.js';
export * from './hooks/index.js';
export * from './lazy/index.js';
export * from './memory/index.js';
export * from './middleware/index.js';
export * from './runtime/index.js';
export * from './signals/index.js';
export * from './types.js';
export * from './utility/index.js';
