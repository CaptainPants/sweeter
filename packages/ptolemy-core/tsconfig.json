{
    "extends": "../../tsconfig.base.json",
    "include": [
        "src/**/*.ts",
        "src/**/*.tsx"
    ],
    "compilerOptions": {
        "rootDir": "src",
        "lib": [
            "ES2022",
            "DOM" // DOM used, but this library is intended to be consumable by non-DOM libraries so special care to be taken
        ],
        "noEmit": true,
        "paths":{
        }
    }
}
