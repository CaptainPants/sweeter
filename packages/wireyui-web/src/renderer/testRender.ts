import { WebRenderer } from './WebRenderer.js';

export interface TestRenderResult {
    readonly nodes: NodeListOf<ChildNode>;
    readonly getHTML: () => string;
    readonly dispose: () => void;
}

/**
 * Convenience function for rendering isolated JSX chunks for unit tests
 * @param callback
 * @returns
 */
export function testRender(callback: () => JSX.Element): TestRenderResult {
    const renderer = new WebRenderer();

    const root = document.createElement('div');

    const dispose = renderer.start(root, callback);

    return {
        nodes: root.childNodes,
        getHTML: () => root.innerHTML,
        dispose,
    };
}
