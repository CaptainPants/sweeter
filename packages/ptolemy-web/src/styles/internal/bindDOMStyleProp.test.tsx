/* @jsxImportSource ../.. */

import { $mutable } from '@serpentis/ptolemy-core';

import { testRender } from '../../test/testRender.js';

it('bindStyle', () => {
    const color = $mutable('red');

    const res = testRender(() => <div style={{ color: color }}></div>);

    expect(res.getHTML()).toMatchSnapshot();

    color.value = 'green';

    expect(res.getHTML()).toMatchSnapshot();

    res.dispose();
});
