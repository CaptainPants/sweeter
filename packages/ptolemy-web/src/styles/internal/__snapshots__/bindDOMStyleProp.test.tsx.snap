// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bindStyle 1`] = `"<div style="color: red;"></div>"`;

exports[`bindStyle 2`] = `"<div style="color: green;"></div>"`;
